GitCommands/Git/IndexLockManager.cs

```diff
@@ -100,7 +95,7 @@ private void DeleteIndexLock(string fileName)
             }
             catch (Exception ex)
             {
-                throw new FileDeleteException(fileName, ex);
+                throw new ExternalOperationException(command: "File.Delete", arguments: fileName, FileSystemUtility.GetWorkingDirectoryNoEx(), ex);
```

Comment:
I struggle to see any value in `command: "File.Delete"` for a user. Also `Directory.GetCurrentDirectory()` isn't appropriate here, it must be the the current repo working directlry.
```suggestion
                throw new ExternalOperationException(command: null, arguments: fileName, workingDir, ex)
```

Reply:
>  Also Directory.GetCurrentDirectory() isn't appropriate here, it must be the the current repo working directlry.

I cannot agree. Filesystem operations are relative to the CWD not to the repo.
The CWD does not matter only if the `argument` contains the full path.
The repo path - or the worktree folder in case of the screenshot - is contained in the `argument`.

> I struggle to see any value in command: "File.Delete" for a user.

As already proposed, these strings could be translated by the `BugReporter`, if you do not want translatable strings in the `FileSystem` wrapper to be created.
I did not want to put too much into one PR.